BUILDING.md

```diff
@@ -0,0 +1,92 @@
+# Building thin-edge.io
+
+## Installing tools
+
+### Rust toolchain
+
+To install Rust please follow [Official installation guide](https://www.rust-lang.org/tools/install).
+To get started you need Cargo's bin directory (`$HOME/.cargo/bin`) in your PATH
+environment variable.
+
+```shell
+export PATH=$PATH:$HOME/.cargo/bin
+```
+
+And then you can run `rustc` to view current version:
+
+```shell
+$ rustc --version
+rustc 1.51.0 (2fd73fabe 2021-03-23)
+```
+
+> Note: Above command will add rust to path only for existing session,
+> after you restart the session you will have to add it again,
+> to add rust to the path permanently it will depend on your shell but for Bash,
+> you simply need to add the line from above, `export PATH=$PATH:$HOME/.cargo/bin` to your ~/.bashrc.
+
+> For other shells, you'll want to find the appropriate place to set a configuration at start time,
+> eg. zsh uses ~/.zshrc. Check your shell's documentation to find what file it uses.
+
+thin-edge.io operates the `MSRV` (Minimum Supported Rust Version) and uses stable toolchain.
+
+Current MSRV is `1.46.0`.
+
+## Compiling
+
+To build thin-edge.io we are using `cargo`.
+
+As we are using a cargo workspace for all our crates all binaries are put in `./target/` directory with target's name eg: `./target/debug` or `./target/armv7-unknown-linux-gnueabihf` dependent on the type of build.
+
```

Comment:
is not it ./target/debug or ./target/release? then armv7*

Reply:
No, for cross builds it goes to `./target/arch_name/profile_name`.
Maybe I should just add separate section for cross compilation to not introduce confusion.

Reply:
Using `<>` for architecture may help people to understand slightly better? 
`./target/<architecture>/debug`, `./target/<architecture>/release`